src/hidpp20.c

```diff
@@ -1129,6 +1129,7 @@ hidpp20_onboard_profiles_read_sector(struct hidpp20_device *device,
 	count = sector_size;
 
 	for (offset = 0; offset < sector_size; offset += 16) {
+		offset = (sector_size - offset < 16) ? sector_size - 16 : offset;
```

Comment:
Just because it took me a while to understand this, I'd like to have a comment here explaining why we go backward in the reading.

Something like `if the sector size is not a multiple of 16, we force reading the last 16 bytes, even if it means re-reading part of the previous chunk`.

(feel free to amend this)

Also, doesn't the write request a similar change?

Reply:
For writes, the total count of bytes we are writing is given to the mouse when issuing the MEMORY_ADDR_WRITE command. So the firmware knows how many bytes are relevant in the last MEMORY_WRITE message. As for the comment, coming right up!